### ISSUE TITLE
BenchmarkResult plots broken!

### ISSUE THREAD

**Issue by giuseppec** [mlr-org/mlr#1299]
``` r
set.seed(1)
lrn = list(
  makeLearner("classif.rpart"), 
  makeBaggingWrapper(makeLearner("classif.rpart", cp = 0), bw.iter = 100), 
  makeLearner("classif.randomForest", ntree = 100)
)
bmr = benchmark(lrn, binaryclass.task)

plotBMRRanksAsBarChart(bmr)
plotBMRBoxplots(bmr)
plotBMRSummary(bmr)
```
1. The legend does not show the names for wrapped learners in all three plots (however, it seems to work when `pretty.names = FALSE`)
2. The colours/ranks for `plotBMRRanksAsBarChart` seem to be wrong, i.e., randomForest seems to be rank 3 although it has the lowest mmce value.

**Comment by giuseppec** [comment]
Another strange observation:

The `pretty.names` arg for `plotBMRBoxplots` does something completely different than the `pretty.names` arg for the other BMR plots (i.e., it prettifies the measure and not the learner names). Although it is correctly documented I asked myself why we don't simply also "prettify" the learner names here?

**Comment by berndbischl** [comment]
@giuseppec 
can you try to fix this in a PR pls?
@larskotthoff @zmjones 
can we somehow unit test this now? please say yes....

**Comment by larskotthoff** [comment]
Yes, we can test this by checking that the learner names (or IDs, or whatever) appear in the generated SVG. We can't easily check whether they appear in the correct place (which shouldn't be an issue here).

**Comment by giuseppec** [comment]
The problem with the first issue is that we don't have a `short.name` slot for wrapped learners. Should we define a `getLearnerShortName` to solve this?

**Comment by schiffner** [comment]
What would this `getLearnerShortName` do exactly?

**Comment by giuseppec** [comment]
For a simple learner it would just return the `learner$short.name` . For a (multiple) wrapped learner, it would successively attach the additional string of the wrapper (which is also attached to the learner id, e.g., `.bagged` for the bagging wrapper and `.imputed` for the impute wrapper etc.).

Another alternative would be to modify each single mlr wrapper and add a `$short.name` slot to all wrappers. Due to laziness, I would prefer the `getLearnerShortName` solution.

**Comment by giuseppec** [comment]
Yet another (more simpler) solution would be to remove the `pretty.names` option completely as this is also the source of the second issue mentioned in my first post (basically  `pretty.names=TRUE` rewrites the legend of `plotBMRRanksAsBarChart` without considering the correct order).

**Comment by schiffner** [comment]
My 2 cents are:
At the moment I like the pretty.names option because it's a quick and convenient way to get ok looking labels and I would maintain it as long as it is not a burden (I think the order is very easy to fix by setting `use.names = TRUE` in `getBMRLearnerShortNames`, but I haven't tried it out properly.).
I would be pro removing the pretty.names option in favor of convenience functions to rename stuff in container objects like BenchmakResults, ResampleResults etc.

**Comment by larskotthoff** [comment]
`getLearnerShortName` sounds good and, IIRC, would remove the need for the `pretty.names` option.

**Comment by schiffner** [comment]
I think this isn't solved completely yet.
I just looked at the tutorial (http://mlr-org.github.io/mlr-tutorial/devel/html/benchmark_experiments/index.html#benchmark-analysis-and-visualization), and
`plotBMRRanksAsBarChart` shows incorrect results.

**Comment by schiffner** [comment]
Here is what I found out:
* `plotBMRBoxplots` gives wrong results if `order.lrns` is used in conjunction with `pretty.names`.
* `plotBMRSummary` ignores the `pretty.names` option. This is just due to a typo in the code (`learner.id` vs. `learners.id`). (And we need to test if that's possible that the short names and not the ids appear in the plot.)
* The labels in `plotBMRRanksAsBarChart` are still wrong if the `pretty.name` option is used.
* The way how `plotBMRRanksAsBarChart ` was fixed breaks the `order.lrns` option.

Most problems are caused by lines like
`levels(df$learner.id) = getBMRLearnerShortNames(bmr)`
which don't take into account that the factor levels might already have an ordering
(alphabetically or through `order.lrns`).

I can take care of this next week, unless someone else wants to do this. :)

**Comment by larskotthoff** [comment]
Would be great if you could have a look at this!

**Comment by mllg** [comment]
Maybe this might help for unit testing:

https://github.com/lionel-/vdiffr

**Comment by giuseppec** [comment]
@florianfendt  you could use something like

```r
  df = as.data.frame(convertBMRToRankMatrix(bmr, measure, ties.method = ties.method, aggregation = aggregation))
  if (pretty.names) {
    learner.id = as.factor(getBMRLearnerShortNames(bmr))
  } else {
    learner.id = as.factor(rownames(df))
  }
  df$learner.id = learner.id
```

to ensure that the order is correct, however the `orderBMRLrns` seem still to have no effect on the plot order, you need to check this and also check what happens with orderBMRLrns when pretty.names is TRUE/FALSE.

**Comment by larskotthoff** [comment]
This should be resolved in #1354.
